File: 백준/Silver/18258. 큐 2/큐 2.py
from collections import deque

n = int(input())
queue = deque()
answer = []
for _ in range(n):
    temp = -1
    commend = input().split(" ")
    if len(commend) == 2:
        queue.append(commend[1])
    else:
        if commend[0] == 'pop':
            if queue:
                temp = queue.popleft()
        elif commend[0] == 'size':
            temp = len(queue)
        elif commend[0] == 'empty':
            if not queue:
                temp = 1
            else:
                temp = 0
        elif commend[0] == 'front':
            if queue:
                temp = queue[0]
        else:
            if queue:
                temp = queue[-1]
        answer.append(temp)

for x in answer:
    print(x)

File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();
        int[] queue = new int[n];
        int head = 0; int tail = 0;
        StringTokenizer st;
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            String commend = st.nextToken();
            int x ;
            if (commend.equals("push")) {
                    x = Integer.parseInt(st.nextToken());
                    queue[tail++] = x;
            } else if (commend.equals("pop")) {
                if (head == tail) sb.append(-1).append("\n");
                else {
                    sb.append(queue[head]).append("\n");
                    head++;
                }
            } else if (commend.equals("size")) {
                sb.append(tail - head).append("\n");
            } else if (commend.equals("empty")) {
                if (head == tail) sb.append("1").append("\n");
                else sb.append(0).append("\n");
            } else if (commend.equals("back")) {
                if (head == tail) sb.append(-1).append("\n");
                else sb.append(queue[tail - 1]).append("\n");
            } else if (commend.equals("front")) {
                if (head == tail) sb.append(-1).append("\n");
                else sb.append(queue[head]).append("\n");
            }
        }
        System.out.println(sb);
    }
}
